src/BookAnimation.ts

```diff
@@ -14,11 +14,14 @@ function roundToTick(tickSize: number, price: number): number {
     return tickSize * Math.round(price / tickSize)
 }
 
-// Surely must be a better mathematical way?
-function getPrecision(num: number) {
-    const strs = num.toString().split('.')
-    if(strs.length !== 2) return 0
-    return strs[1].length
+export function getPrecision(num: number) {
```

Comment:
Good spot with the `1e-7` stuff, I thought I had put something in to get around it but looks like I either forgot it, or lost it somehow :/

There is still an issue with this however. The approach you used was one I used the first time I tried this, but unfortunately it did not work. The reason for this becomes apparent if you try calling `getPrecision` with e.g. `0.025`. The method you propose gives a precision of 2, when in fact it really needs a precision of 3.

I pushed a fix for the string based one for now, but I do think there must be a nicer way. I tried a looping solution but it was a bit janky.